hydromt/raster.py

```diff
@@ -2233,6 +2234,243 @@ def tile_window(shape, px):
         with open(join(root, f"{mName}.yml"), "w") as f:
             yaml.dump({mName: yml}, f, default_flow_style=False, sort_keys=False)
 
+    def to_osm_tiles(
+        self,
+        root: str,
+        zoom_levels: list,
+        bbox: tuple = (),
+        method: str = "nearest",
+        min_lvl: int = 5,
+        driver="GTiff",
+        **kwargs,
+    ):
+        """Export rasterdataset to tiles in an osm structure.
+
+        Parameters
+        ----------
+        root : str
+            Path where the database will be saved
+            Database yml will be put one directory above
+        zoom_levels : list
+            Zoom levels to be put in the database
+        bbox : tuple, optional
+            A region where the tiles should be generated for
+            (if absent, the entire input raster dataset will be tiled)
+        method : str, optional
+            How to resample the data when downscaling.
+            E.g. 'nearest' for resampling with the nearest value
+        min_lvl : int, optional
+            Minimal starting level from to where to downscale from.
+            Increase if memory issues are encountered.
+        driver : str, optional
+            GDAL driver (e.g., 'GTiff' for geotif files), or 'netcdf4' for netcdf files.
+        **kwargs
+            Key-word arguments to write raster files
+        """
+        resampling = getattr(Resampling, method, None)
+        if resampling is None:
+            raise ValueError(f"Resampling method unknown: {method}.")
+        px_size = 256  # Fixed resolution for osm
+
+        assert self._obj.ndim == 2, "Only 2d datasets are accepted..."
+        obj = self._obj.copy()
+        obj = obj.transpose(self.y_dim, self.x_dim)
+
+        mName = os.path.normpath(os.path.basename(root))
+
+        def create_folder(path):
+            if not os.path.exists(path):
+                os.makedirs(path)
+
+        def transform_res(dres, transformer):
+            return transformer.transform(0, dres)[0]
```

Comment:
This function is made to transform coordinates, not resolution. For this (and also to calculate the initial zoom level, see other comment) we could make use of e.g. [rasterio.warp.calculate_default_transform](https://rasterio.readthedocs.io/en/latest/api/rasterio.warp.html#rasterio.warp.calculate_default_transform).

Reply:
Agreed, this was a quick but bad solution.

Reply:
Fixed that.